src/components/App/index.spec.tsx

```diff
@@ -161,4 +162,46 @@ describe(__filename, () => {
 
     expect(dispatch).not.toHaveBeenCalled();
   });
+
+  it('does not render application errors when there is no error', () => {
+    const root = render();
+
+    expect(root.find(`.${styles.errors}`)).toHaveLength(0);
+  });
+
+  it('renders application errors', () => {
+    const error1 = new Error('error 1');
+    const error2 = new Error('error 2');
+    const store = configureStore();
+    store.dispatch(errorsActions.showError({ error: error1 }));
+    store.dispatch(errorsActions.showError({ error: error2 }));
+
+    const root = render({ store });
+
+    expect(root.find(`.${styles.errors}`)).toHaveLength(1);
+    expect(root.find(Alert)).toHaveLength(2);
+    expect(root.find(Alert).at(0)).toIncludeText(error1.message);
+    expect(root.find(Alert).at(1)).toIncludeText(error2.message);
+  });
+
+  it('dispatches dismissError() when dismissing an error', () => {
```

Comment:
I don't know. It is dismissed by ID, which is checked in this test case. I can add one more test case if you feel that would be better.

Reply:
My concern is that the `id` being checked is using information from the state (`errors[errors.length - 1]`), rather than information from what was rendered on the page, so an error in the logic of the page might not be caught. Does that make sense?